gnocchi/rest/api.py

```diff
@@ -1655,17 +1655,18 @@ def groupper(r):
 # FIXME(sileht): should be in aggregates.api but we need to split all
 # controllers to do this
 def validate_qs(start, stop, granularity, needed_overlap, fill):
-    try:
-        needed_overlap = float(needed_overlap)
-    except ValueError:
-        abort(400, {"cause": "Argument value error",
-                    "detail": "needed_overlap",
-                    "reason": "Must be a number"})
-    if needed_overlap != 100.0 and start is None and stop is None:
-        abort(400, {"cause": "Argument value error",
-                    "detail": "needed_overlap",
-                    "reason": "start and/or stop must be provided "
-                    "if specifying needed_overlap"})
+    if needed_overlap is not None:
+        try:
+            needed_overlap = float(needed_overlap)
+        except ValueError:
+            abort(400, {"cause": "Argument value error",
+                        "detail": "needed_overlap",
+                        "reason": "Must be a number"})
+        if needed_overlap != 100.0 and start is None and stop is None:
```

Comment:
tbh, not related to this patch, but start/stop don't need to be provided. i only added it originally because the previous logic made the start/stop boundaries ambiguous. we have test now that explicitly say it's the first/last points where all the series overlap.

Reply:
yay go for another patch @chungg :)